sktime/forecasting/autots.py

```diff
@@ -0,0 +1,650 @@
+    generation_timeout (int):
+        The timeout for each generation. if not None, this is the number of minutes
+        from start at which the generational search ends, then proceeding to
+        validation This is only checked after the end of each generation, so
+        only offers an 'approximate' timeout for searching. It is an overall
+        cap for total generation search time, not per generation.
+    current_model_file (str):
+        The file containing the current model.
+        file path to write to disk of current model params
+        (for debugging if computer crashes). .json is appended
+    verbose (int):
+        The verbosity level.
+        setting to 0 or lower should reduce most output.
+        Higher numbers give more output.
+    n_jobs (int):
+        The number of jobs to run in parallel.
+        Number of cores available to pass to parallel processing.
+        A joblib context manager can be used instead (pass None in this case).
+        Also 'auto'.
+    """
+
+    _tags = {
+        "scitype:y": "univariate",
+        "authors": ["winedarksea", "MBristle"],  # winedarksea for autots library
+        "maintainers": ["MBristle"],
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "ignores-exogeneous-X": True,  # TODO: add capability
+        "capability:insample": False,
+        "capability:pred_int:insample": False,
+        "capability:pred_int": False,  # TODO: add capability
+        "requires-fh-in-fit": True,
+        "python_dependencies": ["autots", "pandas", "statsmodels", "scipy"],
+        "python_version": ">=3.6",
+    }
+
+    def __init__(
+        self,
+        model_name: str = "",
+        model_list: list = "superfast",
+        frequency: str = "infer",
+        prediction_interval: float = 0.9,
+        max_generations: int = 10,
+        no_negatives: bool = False,
+        constraint: float = None,
+        ensemble: str = "auto",
+        initial_template: str = "General+Random",
+        random_seed: int = 2022,
+        holiday_country: str = "US",
+        subset: int = None,
+        aggfunc: str = "first",
+        na_tolerance: float = 1,
+        metric_weighting: dict = None,
+        drop_most_recent: int = 0,
+        drop_data_older_than_periods: int = 100000,
+        transformer_list: dict = "auto",
+        transformer_max_depth: int = 6,
+        models_mode: str = "random",
+        num_validations: int = "auto",
+        models_to_validate: float = 0.15,
+        max_per_model_class: int = None,
+        validation_method: str = "backwards",
+        min_allowed_train_percent: float = 0.5,
+        remove_leading_zeroes: bool = False,
+        prefill_na: str = None,
+        introduce_na: bool = None,
+        preclean: dict = None,
+        model_interrupt: bool = True,
+        generation_timeout: int = None,
+        current_model_file: str = None,
+        verbose: int = 1,
+        n_jobs: int = -2,
```

Comment:
I like n_jobs=0.5 (which translates as 50% of available cores) because I have found that pushing above 75% of used cores is more likely to lead to crashes (it really depends on OS, size of memory, size of data, which models are called). So n_jobs=-2  as set here (so 1 less than the max cores) will probably be stable on most computers, but lower is safer.

Generally things are pretty compatible across generations, so you might be able to go back earlier than 0.6.x. The only challenge I see here is if new args get added to the AutoTS() class, and you decide to add them here. Might it be possible to allow `**kwargs` here and passing through, instead of hard coding all the AutoTS args?

It might be nice to have some way to import a template (.import_template and .export_template) without going through the base modelclass. Using a saved model template is pretty helpful for avoiding long model searches and getting good models fast.

Otherwise looks okay.

Reply:
answered in bulk below, feel free to continue this thread